types/node_key_test.go

```diff
@@ -7,12 +7,22 @@ import (
 
 	"github.com/stretchr/testify/require"
 
-	tmrand "github.com/tendermint/tendermint/libs/rand"
 	"github.com/tendermint/tendermint/types"
 )
 
+func mustTempPath(t *testing.T, name string) string {
+	t.Helper()
+
+	dir, err := os.MkdirTemp("", t.Name()+"*")
```

Comment:
Re: concatenation. We're really not formatting anything, so I think it's more readable this way, but if you feel strongly I'll change it.

Re: /tmp. The test harness respects `$TMPDIR` correctly, and on ~all Linux implementations, `/tmp` is auto-cleaned at boot, so I think we're safe. Our tests are generally _terrible_ about cleaning up after themselves, though.

Reply:
`fmt.Sprint` (sans `f`) just use's the fmt package's buffer pool and calls Stringer/etc stuff, and I feel like I've been burned by concatenation not using pooled resources enough times that I'm just instinctively wary. 

Re, the test harness, I think it's fine, I guess I'm more worried about: 
- machines that don't reboot between test runs (or that run multiple tests,) 
- I wish that we (as an industry) generally set TMPDIR to not `/tmp` because `/tmp` is ram backed on a lot of systems and it'd 
- my general distrust of cleanup actually happening. (given ooms/etc. I see a lot of things leak...)

Like I said, it's totally fine, and probably a case where I've been burned in the remote past and am more touchy on these points than I should be otherwise.

Reply:
Re: Sprint. Yup, and I like that for arbitrary values. But we know the type is a string in this case. (Mooted anyway, since Callum's suggestion was simpler 🙂)

I agree with all of that, re: tests. That said: This is now one of the few of our tests that makes _any_ effort to clean up after itself. I like to do that more—my $TMPDIR is littered with hundreds of garbage directories written by our tests, and as you say, one doesn't necessarily reboot often.